app/code/core/Mage/Core/Model/Encryption.php

```diff
@@ -153,53 +148,53 @@ public function validateHashByVersion(#[\SensitiveParameter] $password, $hash, $
         return hash_equals($this->hash($salt . $password, $version), $hash);
     }
 
-    /**
-     * Instantiate crypt model
-     *
-     * @param string $key
-     * @return Varien_Crypt_Mcrypt
-     */
-    protected function _getCrypt($key = null)
+    public function encrypt(#[\SensitiveParameter] string $data): string
     {
-        if (!$this->_crypt) {
-            if ($key === null) {
-                $key = (string) Mage::getConfig()->getNode('global/crypt/key');
-            }
-            $this->_crypt = Varien_Crypt::factory()->init($key);
-        }
-        return $this->_crypt;
+        $key = Mage::getEncryptionKeyAsBinary();
+        $nonce = random_bytes(SODIUM_CRYPTO_SECRETBOX_NONCEBYTES);
+        $ciphertext = sodium_crypto_secretbox($data, $nonce, $key);
+        $encrypted = sodium_bin2base64($nonce . $ciphertext, SODIUM_BASE64_VARIANT_ORIGINAL);
+        sodium_memzero($data); // Clean sensitive data from memory
+        return $encrypted;
     }
 
-    /**
-     * Encrypt a string
-     *
-     * @param string $data
-     * @return string
-     */
-    public function encrypt($data)
+    public function decrypt(#[\SensitiveParameter] string $data): string
     {
-        return base64_encode($this->_getCrypt()->encrypt((string) $data));
-    }
+        try {
+            $decoded = sodium_base642bin($data, SODIUM_BASE64_VARIANT_ORIGINAL);
+        } catch (SodiumException $e) {
+            $exception = new Exception('Invalid base64 encoding: ' . $e->getMessage());
+            Mage::logException($exception);
+            return '';
+        }
 
-    /**
-     * Decrypt a string
-     *
-     * @param string $data
-     * @return string
-     */
-    public function decrypt($data)
-    {
-        return str_replace("\x0", '', trim($this->_getCrypt()->decrypt(base64_decode((string) $data))));
+        if (strlen($decoded) < (SODIUM_CRYPTO_SECRETBOX_NONCEBYTES + SODIUM_CRYPTO_SECRETBOX_MACBYTES)) {
+            $exception = new Exception('Data is too short to be valid');
+            Mage::logException($exception);
+            return '';
+        }
+
+        $key = Mage::getEncryptionKeyAsBinary();
+        $nonce = mb_substr($decoded, 0, SODIUM_CRYPTO_SECRETBOX_NONCEBYTES, '8bit');
+        $ciphertext = mb_substr($decoded, SODIUM_CRYPTO_SECRETBOX_NONCEBYTES, null, '8bit');
+        $plaintext = sodium_crypto_secretbox_open($ciphertext, $nonce, $key);
+
+        if ($plaintext === false) {
+            $exception = new Exception('Decryption failed: data may be corrupted or tampered with');
+            Mage::logException($exception);
+            return '';
+        }
+
+        // Clean sensitive data from memory
+        sodium_memzero($decoded);
+        sodium_memzero($nonce);
+        sodium_memzero($ciphertext);
```

Comment:
```suggestion
        sodium_memzero($data);
        sodium_memzero($decoded);
        sodium_memzero($key);
        sodium_memzero($nonce);
        sodium_memzero($ciphertext);
```

A few more values here too.